src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -37,34 +38,46 @@ public class Connection : IConnection
         public event Action<byte[]> Received;
         public event Action<Exception> Closed;
 
-        public Connection(Uri url)
-            : this(url, null)
+        public HttpConnection(Uri url)
+            : this(url, TransportType.WebSockets)
         { }
 
-        public Connection(Uri url, ILoggerFactory loggerFactory)
+        public HttpConnection(Uri url, TransportType transportType)
+                    : this(url, transportType, loggerFactory: null)
+        {
+        }
+
+        public HttpConnection(Uri url, ILoggerFactory loggerFactory)
+            : this(url, TransportType.WebSockets, loggerFactory, httpMessageHandler: null)
+        {
+        }
+
+        public HttpConnection(Uri url, TransportType transportType, ILoggerFactory loggerFactory)
+            : this(url, transportType, loggerFactory, httpMessageHandler: null)
+        {
+        }
+
+        public HttpConnection(Uri url, TransportType transportType, ILoggerFactory loggerFactory, HttpMessageHandler httpMessageHandler)
         {
             Url = url ?? throw new ArgumentNullException(nameof(url));
 
             _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
-            _logger = _loggerFactory.CreateLogger<Connection>();
+            _logger = _loggerFactory.CreateLogger<HttpConnection>();
+            _httpClient = httpMessageHandler == null ? new HttpClient() : new HttpClient(httpMessageHandler);
+            _transportFactory = new DefaultTransportFactory(transportType, _loggerFactory, _httpClient);
         }
 
-        public Task StartAsync() => StartAsync(transportFactory: null, httpClient: null);
-        public Task StartAsync(HttpClient httpClient) => StartAsync(transportFactory: null, httpClient: httpClient);
-        public Task StartAsync(ITransportFactory transportFactory) => StartAsync(transportFactory, httpClient: null);
-        public Task StartAsync(TransportType transportType) => StartAsync(transportType, httpClient: null);
-        public Task StartAsync(TransportType transportType, HttpClient httpClient)
+        public HttpConnection(Uri url, ITransportFactory transportFactory, ILoggerFactory loggerFactory, HttpMessageHandler httpMessageHandler)
         {
-            if (httpClient == null)
-            {
-                // We are creating the client so store it to be able to dispose
-                _httpClient = httpClient = new HttpClient();
-            }
+            Url = url ?? throw new ArgumentNullException(nameof(url));
```

Comment:
Yes, you copied it and have not changed it and also left one that was there - so you are progressing ;)